### PR TITLE
Make rand_tangent on adjoint an transpose return natural

### PR DESCRIPTION
**Pull request by oxinabox**
closes https://github.com/JuliaDiff/FiniteDifferences.jl/issues/151

### CODE REVIEW

File: src/rand_tangent.jl [hunk 1]
```diff
@@ -13,11 +13,13 @@ rand_tangent(rng::AbstractRNG, x::Integer) = NoTangent()
...
 rand_tangent(rng::AbstractRNG, x::T) where {T<:Number} = randn(rng, T)
 
-# TODO: right now Julia don't allow `randn(rng, BigFloat)` 
+# TODO: right now Julia don't allow `randn(rng, BigFloat)`
 # see: https://github.com/JuliaLang/julia/issues/17629
 rand_tangent(rng::AbstractRNG, ::BigFloat) = big(randn(rng))
 
 rand_tangent(rng::AbstractRNG, x::StridedArray) = rand_tangent.(Ref(rng), x)
+rand_tangent(rng::AbstractRNG, x::Adjoint) = adjoint(rand_tangent(rng, parent(x)))
+rand_tangent(rng::AbstractRNG, x::Tangent) = tangent(rand_tangent(rng, parent(x)))
```
Comment (nickrobinson251) [inline src/rand_tangent.jl:23]: Typo?

File: src/rand_tangent.jl [hunk 2]
```diff
@@ -13,11 +13,13 @@ rand_tangent(rng::AbstractRNG, x::Integer) = NoTangent()
...
```
Comment (oxinabox) [inline src/rand_tangent.jl:22]:
```suggestion
rand_tangent(rng::AbstractRNG, x::Transpose) = transpose(rand_tangent(rng, parent(x)))
```

File: test/rand_tangent.jl [hunk 3]
```diff
@@ -24,6 +24,11 @@ using FiniteDifferences: rand_tangent
         (randn(Complex{Float32}, 5, 4), Matrix{Complex{Float32}}),
         ([randn(5, 4), 4.0], Vector{Any}),
 
+        # Wrapper Arrays
+        (randn(5, 4)', Adjoint{Float64, Matrix{Float64}},
+        (transpose(randn(5, 4)), Transpose{Float64, Matrix{Float64}},
```
Comment (oxinabox) [inline test/rand_tangent.jl:29]:
```suggestion
        (randn(5, 4)', Adjoint{Float64, Matrix{Float64}}),
        (transpose(randn(5, 4)), Transpose{Float64, Matrix{Float64}}),
```

**Comment by oxinabox** [comment]
Added two more changes.

Firstly to stop testing exacty what type `+` returned.
Because `Adjoint  + Adjoint = Matrix` so it isn't trivially consistent, but per my last comments in #151 I don't really care that much, we have to handle that case because we have to handle that case.
Better to be pragmatic.
It's also out of the power of this package to fix.
That will be fixed with projection/rerepresentation which will be defined else where.

Secondly, this really relaxed a lot about how difference works.
I think the new behavour is intuitive but it feels more dubious.
But I want to delete `difference` anyway, so i don't think it matters (#166)

**Comment by mzgubic** [comment]
LGTM after `difference` is deleted

**Review by mzgubic — APPROVED** [review]

**Pull request merged**